### PR TITLE
Reorganize the list of data files for caching

### LINKED ISSUE #3144: Reorganize the list of cache files used in CI
Many remote files are cached as GitHub Actions assets to be reused in CI workflows. These files are listed in the `pygmt/helpers/caching.py` file. See below:

https://github.com/GenericMappingTools/pygmt/blob/f0d4d21f7518b4dc10f503ec26e2e5b603d8891e/pygmt/helpers/caching.py#L13-L20

Currently, the list is a little messy. Some potential improvements:

1. Put some comments (e.g., "Names like @N35E135.earth_relief_03s_g.nc are for internal use only.") at the start of the list. 
2. Remove some comments like "# Specific grid for 15s test"
3. Update the order of some entries. First categorize files into groups and sort alphabetically in each group.

Helps are welcome.

### PR DESCRIPTION
**Pull request by seisman**
~~Due to a likely upstream bug, we can't request multiple tiled grids in a single `which` call (xref: #3170). We have to call `which` multiple times and download a single tiled grid in each call, to make the "Cache data" workflow pass again.~~

~~I also take this opportunity to address the request in #3144.~~

Closes #3144.

Edit: With the workaround in PR #3174, now it's OK to download multiple tiled grids. So, this PR only reorganize the list of data files for caching to make it easy to maintain.

### CODE REVIEW

File: pygmt/helpers/caching.py [hunk 1]
```diff
@@ -100,4 +89,5 @@ def cache_data():
         "@tut_ship.xyz",
         "@usgs_quakes_22.txt",
     ]
-    which(fname=datasets, download="a")
+
+    which(fname=datasets + caches + tiles, download="a")
```
Comment (weiji14) [inline pygmt/helpers/caching.py:93]: Can we continue to use just a single list?

Comment (seisman) [reply pygmt/helpers/caching.py:93]: We can. I just feel that having three separate lists makes it easier to organize the file list.

File: pygmt/helpers/caching.py [hunk 2]
```diff
@@ -9,89 +9,78 @@ def cache_data():
...
+        "@earth_age_01d_g",
+        "@earth_day_01d_p",
+        "@earth_faa_01d_g",
         "@earth_gebco_01d_g",
         "@earth_gebcosi_01d_g",
         "@earth_gebcosi_15m_p",
-        "@earth_relief_01d_p",
+        "@earth_geoid_01d_g",
+        "@earth_mag4km_01d_g",
+        "@earth_mag_01d_g",
```
Comment (weiji14) [inline pygmt/helpers/caching.py:22]:
```suggestion
        "@earth_mag_01d_g",
        "@earth_mag4km_01d_g",
```

File: pygmt/helpers/caching.py (hunk 1 above)
Comment (weiji14) [reply pygmt/helpers/caching.py:93]: Well, codecov is warning that there's missing code coverage for the extra two lists :slightly_smiling_face:

File: pygmt/helpers/caching.py [hunk 3]
```diff
@@ -9,89 +9,78 @@ def cache_data():
...
+        "@N00W090.earth_relief_03m_p.nc",
+        "@N00E135.earth_relief_30s_g.nc",
+        "@N00W010.earth_relief_15s_p.nc",
+        "@N30W120.earth_relief_15s_p.nc",
+        "@N04W010.earth_relief_03s_g.nc",
+        "@N35E135.earth_relief_03s_g.nc",
+        "@N37W120.earth_relief_03s_g.nc",
+        "@S15W105.earth_synbath_30s_p.nc",
+        "@S90E000.earth_wdmam_03m_g.nc",
+        "@N00W030.earth_vgg_01m_p.nc",
```
Comment (weiji14) [inline pygmt/helpers/caching.py:59]:
```suggestion
        "@N00W030.earth_vgg_01m_p.nc",
        "@S90E000.earth_wdmam_03m_g.nc",
```

File: pygmt/helpers/caching.py (hunk 1 above)
Comment (seisman) [reply pygmt/helpers/caching.py:93]: That's a good point to push me to merge them into a single list.

Comment (seisman) [reply pygmt/helpers/caching.py]: Done in 72fcaa1b18ddb922005c0dc482788d0e439098cc.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**